src/storage_persister.rs

```diff
@@ -75,8 +76,59 @@ impl StoragePersister {
         result
     }
 
-    pub fn read_channel_manager(&self, _user_config: UserConfig) {
-        // TODO: Implement
+    #[allow(clippy::too_many_arguments, clippy::type_complexity)]
+    pub fn read_channel_manager<M, T, F, L>(
+        &self,
+        chain_monitor: Arc<M>,
+        broadcaster: Arc<T>,
+        keys_manager: Arc<KeysManager>,
+        fee_estimator: Arc<F>,
+        logger: Arc<L>,
+        channel_monitors: Vec<&mut ChannelMonitor<InMemorySigner>>,
+        user_config: UserConfig,
+        chain_params: ChainParameters,
+    ) -> Result<(Option<BlockHash>, SimpleArcChannelManager<M, T, F, L>), InitializationError>
+    where
+        M: Watch<InMemorySigner>,
+        T: BroadcasterInterface,
+        F: FeeEstimator,
+        L: Logger,
+    {
+        if self
+            .storage
+            .object_exists(OBJECTS_BUCKET.to_string(), MONITOR_KEY.to_string())
+        {
+            let data = self
+                .storage
+                .get_object(OBJECTS_BUCKET.to_string(), MONITOR_KEY.to_string());
+            let read_args = ChannelManagerReadArgs::new(
+                keys_manager,
+                fee_estimator,
+                chain_monitor,
+                broadcaster,
+                logger,
+                user_config,
+                channel_monitors,
+            );
+            let mut buffer = Cursor::new(&data);
+            let (block_hash, channel_manager) =
+                <(BlockHash, SimpleArcChannelManager<M, T, F, L>)>::read(&mut buffer, read_args)
+                    .map_err(|e| InitializationError::ChannelMonitorBackup {
+                        message: e.to_string(),
+                    })?;
+            Ok((Some(block_hash), channel_manager))
+        } else {
+            let channel_manager = SimpleArcChannelManager::new(
+                fee_estimator,
+                chain_monitor,
+                broadcaster,
+                logger,
+                keys_manager,
+                user_config,
+                chain_params,
+            );
+            Ok((None, channel_manager))
```

Comment:
Why don't we return the blockhash if it's a new `ChannelManager`? We have it available in `chain_params.best_block.block_hash()`.

Reply:
Ok, I see that you are using this to know if this method is reading or creating a `ChannelManager`. Maybe we can move the "read or create" logic directly to `LightningNode::new()` or create a new function in `lib.rs` that handles that.